src/java/org/apache/cassandra/index/sai/disk/v1/SAICodecUtils.java

```diff
@@ -99,9 +100,26 @@ public static void validate(IndexInput input, long footerPointer) throws IOExcep
         input.seek(current);
     }
 
+    /**
+     * See the {@link org.apache.lucene.codecs.CodecUtil#checksumEntireFile(org.apache.lucene.store.IndexInput)}.
+     * @param input IndexInput to validate.
+     * @throws IOException if a corruption is detected.
+     */
     public static void validateChecksum(IndexInput input) throws IOException
     {
-        CodecUtil.checksumEntireFile(input);
+        IndexInput clone = input.clone();
+        clone.seek(0L);
+        ChecksumIndexInput in = new BufferedChecksumCrc32cIndexInput(clone);
+
+        assert in.getFilePointer() == 0L;
```

Comment:
nit: I guess we might as well throw an error message in here, something like...

```
assert in.getFilePointer() == 0L : in.getFilePointer() + " bytes already read from this input!";
```

Reply:
Fixed.